internal/controllers/clusterrequest_controller.go

```diff
@@ -1172,23 +1179,63 @@ func (t *clusterRequestReconcilerTask) createHwMgrPluginNamespace(
 	return nil
 }
 
+func (t *clusterRequestReconcilerTask) hwMgrPluginNamespaceExists(
+	ctx context.Context, name string) (bool, error) {
+
+	t.logger.InfoContext(
+		ctx,
+		"Plugin: "+fmt.Sprintf("%v", name),
+	)
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	exists, err := utils.DoesK8SResourceExist(ctx, t.client, name, "", namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed check if namespace exists %s: %w", name, err)
+	}
+
+	return exists, nil
+}
+
 func (t *clusterRequestReconcilerTask) createNodePoolResources(ctx context.Context, nodePool *hwv1alpha1.NodePool) error {
+	chkNodepool := &hwv1alpha1.NodePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodePool.Name,
+			Namespace: nodePool.Namespace,
+		},
+	}
+	if exists, err := utils.DoesK8SResourceExist(ctx, t.client, nodePool.Name, nodePool.Namespace, chkNodepool); err != nil {
+		return fmt.Errorf("failed check is nodepool exists %s: %w", nodePool.Name, err)
+	} else if exists {
+		return nil
+	}
 
 	// Create the hardware plugin namespace.
 	pluginNameSpace := nodePool.ObjectMeta.Namespace
-	createErr := t.createHwMgrPluginNamespace(ctx, pluginNameSpace)
-	if createErr != nil {
-		return fmt.Errorf(
-			"failed to create hardware manager plugin namespace %s, err: %w", pluginNameSpace, createErr)
+	if exists, err := t.hwMgrPluginNamespaceExists(ctx, pluginNameSpace); err != nil {
+		return fmt.Errorf("failed check if hardware manager plugin namespace exists %s, err: %w", pluginNameSpace, err)
+	} else if !exists && (pluginNameSpace == utils.TempDellPluginNamespace || pluginNameSpace == utils.UnitTestHwmgrNamespace) {
+		// TODO: For test purposes only. Code to be removed once hwmgr plugin(s) are fully utilized
+		createErr := t.createHwMgrPluginNamespace(ctx, pluginNameSpace)
+		if createErr != nil {
+			return fmt.Errorf(
+				"failed to create hardware manager plugin namespace %s, err: %w", pluginNameSpace, createErr)
+		}
+	} else if !exists {
+		return fmt.Errorf("specified hardware manager plugin namespace does not exist: %s", pluginNameSpace)
 	}
+
 	// Create the clusterInstance namespace.
 	err := t.createClusterInstanceNamespace(ctx, nodePool.GetName())
 	if err != nil {
 		return err
 	}
 
 	// Create/update the node pool resource
-	createErr = utils.CreateK8sCR(ctx, t.client, nodePool, t.object, utils.UPDATE)
+	createErr := utils.CreateK8sCR(ctx, t.client, nodePool, t.object, utils.UPDATE)
```

Comment:
I realized that if you pass an empty string to the "operation". It will not update if already exists
```suggestion
	createErr := utils.CreateK8sCR(ctx, t.client, nodePool, t.object, "")
```

Reply:
@Missxiaoguo thanks for sharing this info with me today!

Reply:
Done.